Fix language selection

Zukye opened the pull request:
This should fix the bug but I cannot reproduce it, so please check.

Commit:
Avoid Null-dereference in getResourceFolderFiles, refs #52.

Used Optional to wrap possible null-values, and use empty Array as fallback, which should avoid NPE

Commit:
Refactor Languages, add Test, refs #52.

Commit:
Add alternative way of languages-loading, should fix #52

DivineThreepwood committed:
Update Language.java

some finals added

DivineThreepwood committed:
Merge branch 'master' into fix-language_selection

DivineThreepwood reviewed (commented):
Your changes seems to solve the "settings not available" issue bco.bcozy#52 , but the language drop down menu has no entries anymore which means switching languages is still not possible.

DivineThreepwood committed:
Merge branch 'master' into fix-language_selection

DivineThreepwood committed:
Merge branch 'master' into fix-language_selection

Commit:
Refactor Languages.

Added file containing all available locales so languages can also be loaded from jar files.
Workaround til a better solution is found.

Commit:
Merge branch 'fix-language_selection' of https://github.com/openbase/bco.bcozy into fix-language_selection

Zukye commented:
So the problem seems to be that you cannot list items inside a jar file. As a workaround I added a file containing a list of available language bundles which would have to be updated every time a new language is added. I couldn't come up with a better solution.

DivineThreepwood committed:
Merge branch 'master' into fix-language_selection

DivineThreepwood committed:
manual merge

DivineThreepwood reviewed (approved):
Thanks for refactoring the language detection.
